src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -1071,29 +1027,31 @@ namespace
 
         try
         {
-            if (extension == ".exr")
-            {
-                write_exr_image(
-                    bf_file_path,
-                    image,
-                    image_attributes);
+            Image transformed_image(image);
+
+            // todo: we may want to be more specific here.
+            if (image.properties().m_channel_count == 4)
+                transform_to_srgb(transformed_image);
+
+            create_parent_directories(bf_file_path);
+
+            const std::string filename = bf_file_path.string();
+            GenericImageFileWriter writer(filename.c_str());
+
+            writer.append_image(&transformed_image);
+
+            writer.set_image_output_format(PixelFormat::PixelFormatUInt8);
```

Comment:
Again, that's incorrect for all HDR formats.

Reply:
set the exported bit depth as
OpenEXR     .exr                                 32-bit
TIFF             .tiff/.tif                             32-bit
RGBE           .hdr                                 32-bit
BMP            .bmp                                 8-bit
JPEG            .jpg/.jpe/.jpeg/.jif/.jfif/.jfi  8-bit